File: Frontend/src/services/government.ts
import { apiService } from './api';
import type {
  AcceptDisasterRequest,
  RejectDisasterRequest,
} from '../types/disaster';
import type {
  CreateResourceRequest,
  UpdateResourceAvailabilityRequest,
  DeleteResourceRequest,
} from '../types/Resource';


class GovernmentService {
  private readonly baseUrl = '/gov';

  async acceptDisaster(request: AcceptDisasterRequest){
    const response = await apiService.post(`${this.baseUrl}/emergency/accept`, request);
    return response.data;
  }

  async rejectDisaster(request: RejectDisasterRequest) {
    const response = await apiService.post(`${this.baseUrl}/emergency/reject`, request);
    return response.data;
  }

    async addResource(request: CreateResourceRequest)  {
    const response = await apiService.post(`${this.baseUrl}/resource/add`, request);
    return response.data;
  }

   async updateResourceAvailability(request: UpdateResourceAvailabilityRequest){
    const response = await apiService.patch(`${this.baseUrl}/resource/update-availability`, request);
    return response.data;
  }

    async deleteResource(request: DeleteResourceRequest){
    const response = await apiService.delete(`${this.baseUrl}/resource/delete`, { data: request });
    return response.data;
  }

}

export const governmentService = new GovernmentService()

File: Backend/app/apis/government.py
from app.models.user import UserProfile
from fastapi import APIRouter, Depends, HTTPException
from app.services.role_service import require_government
from app.services.appwrite_service import AppwriteService
from app.services.second_workflow import create_generate_disaster_task_graph
from app.models.disaster import DisasterRequest
from app.models.resource import ResourcePayload, DeleteResourceRequest, UpdateAvailabilityRequest
from app.models.user import DeleteUser


router = APIRouter(prefix="/gov", tags=["Government"])
appwrite_service = AppwriteService()


@router.post("/emergency/accept")
async def accept_disaster(payload: DisasterRequest, user: UserProfile = Depends(require_government)):
    try:
        appwrite_service.update_disaster_status(payload.disaster_id, "active")
        print(f"Disaster {payload.disaster_id} marked as active by {user.name}")
        graph = create_generate_disaster_task_graph()
        await graph.invoke({"disaster_id": payload.disaster_id})
        
        return {"message": f"Disaster {payload.disaster_id} marked as active."}
    
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Internal error: {str(e)}")

@router.post("/emergency/reject")
async def reject_disaster(payload: DisasterRequest, user: UserProfile = Depends(require_government)):
    try:
        appwrite_service.archive_disaster(payload.disaster_id)
        
        return {"message": f"Disaster {payload.disaster_id} archived."}
    
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Internal error: {str(e)}")

@router.post("/resource/add")
async def add_resource(payload: ResourcePayload, user: UserProfile = Depends(require_government)):
    try:
        appwrite_service.add_resource_to_disaster(payload.disasterId, payload.data)
        
        return {"message": "Resource added successfully"}
    
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Failed to add resource: {str(e)}")
    
@router.delete("/resource/delete")
async def delete_resource(payload: DeleteResourceRequest, user: UserProfile = Depends(require_government)):
    try:
        appwrite_service.delete_resource(payload.resource_id)
        return {"message": f"Resource {payload.resource_id} deleted successfully"}
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Failed to delete resource: {str(e)}")

@router.patch("/resource/update-availability")
async def update_resource_availability(payload: UpdateAvailabilityRequest, user: UserProfile = Depends(require_government)):
    try:
        appwrite_service.update_resource_availability(payload.resource_id, payload.availability)
        return {"message": f"Resource {payload.resource_id} availability updated to {payload.availability}"}
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Failed to update availability: {str(e)}")
    
@router.delete("/user/delete")
async def delete_user(payload: DeleteUser, user: UserProfile = Depends(require_government)):
    try:
        appwrite_service.delete_user(payload.user_id)
        return {"message": f"User {payload.user_id} deleted successfully"}
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Internal error: {str(e)}")

